### DISCUSSION TITLE
Mock Contracts

### DISCUSSION THREAD

**Discussion by vicjuma** [Cyfrin/foundry-full-course-cu#3055] [Q&A] [answered]
Hello, this is on a lighter note😊. Is there anything special about a **mock contract** other than inheritance except this time it is on an interface? With maybe the inability to use _virtual_ and _override_ keywords? Just wondering 😊. Otherwise, 👍 to Patrick and the entire **Cyfrin community**.

**Comment by EngrPips** [comment]
Hello @vicjuma, I don't understand your question. Please break it down more maybe I will understand it.

**Reply by vicjuma** [reply]
@EngrPips it was just an observation because I have reached this point

`contract MockV3Aggregator is AggregatorV3Interface {}`

```
function getAnvilEthConfig() public returns (NetworkConfig memory) {
        if(activeNetworkConfig.priceFeed != address(0)) {
            return activeNetworkConfig;
        }
        // Deploy the mock then return the mock addresses
        vm.startBroadcast();
        MockV3Aggregator mockV3Aggregator = new MockV3Aggregator(NUMBER_OF_DECIMALS, PRICE_OF_ETH_IN_USD);
        vm.stopBroadcast();
        NetworkConfig memory anvilEthConfig = NetworkConfig(address(mockV3Aggregator));
        return anvilEthConfig;
    }
```
Just observed that we are performing **AggregatorV3Interface** interface inheritance on our **MockV3Aggregator** contract. There are no issues though. Everything is 100% awesome.

**Reply by EngrPips** [reply]
Oh, I see. You are just pointing out your observation and not asking a question.

**Reply by vicjuma** [reply]
Yes. It is a great experience😊

**Reply by EngrPips** [reply, marked as answer]
Indeed, Programming is amazing.
